post_processing/ci_plots.jl

```diff
@@ -606,18 +657,126 @@ EDMFBoxPlots = Union{
     Val{:prognostic_edmfx_trmm_column},
 }
 
+"""
+    plot_edmf_vert_profile!(grid_loc, var_group)
+
+Helper function for `make_plots_generic`. Takes a list of variables and plots
+them on the same axis.
+"""
+function plot_edmf_vert_profile!(grid_loc, var_group)
+    z = var_group[1].dims["z"]
+    units = var_group[1].attributes["units"]
+    ax = CairoMakie.Axis(
+        grid_loc[1, 1],
+        ylabel = "z [$(var_group[1].dim_attributes["z"]["units"])]",
+        xlabel = "$(short_name(var_group[1])) [$units]",
+        title = parse_var_attributes(var_group[1]),
+    )
+    for var in var_group
+        CairoMakie.lines!(ax, var.data, z, label = short_name(var))
+    end
+    length(var_group) > 1 && Makie.axislegend(ax)
+end
+
+
+"""
+    plot_parsed_attribute_title!(grid_loc, var)
+
+Helper function for `make_plots_generic`. Plots an OutputVar `var`,
+setting the axis title to `parse_var_attributes(var)`
+"""
+plot_parsed_attribute_title!(grid_loc, var) = viz.plot!(
+    grid_loc,
+    var;
+    more_kwargs = Dict(:axis => ca_kwargs(title = parse_var_attributes(var))),
+)
+
+"""
+    pair_edmf_names(vars)
+
+Groups updraft and gridmean EDMF short names into tuples. 
+Matches on the same variable short name with the suffix "up".
+This assumes that the updraft variable name is the same as the corresponding
+gridmean variable with the suffix "up".
+"""
+function pair_edmf_names(short_names)
+    grouped_vars = Any[]
+    short_names_to_be_processed = Set(short_names)
+
+    for name in short_names
+        name in short_names_to_be_processed || continue
+
+        base_name = replace(name, "up" => "")
+        up_name = base_name * "up"
+
+        if base_name in short_names_to_be_processed &&
+           up_name in short_names_to_be_processed
+            tuple_to_be_added = (base_name, up_name)
+        else
+            tuple_to_be_added = (name,)
+        end
+
+        [pop!(short_names_to_be_processed, n) for n in tuple_to_be_added]
```

Comment:
```suggestion
        foreach(n -> pop!(short_names_to_be_processed, n), tuples_to_be_added)
```
This avoids creating a new vector only to enjoy the side effect

Reply:
And maybe we want to use `delete!` instead of `pop!`